File: 2017/fall/3/sigma0.c
// Sums a range of numbers iteratively

#include <cs50.h>
#include <stdio.h>

int sigma(int m);

int main(void)
{
    int n;
    do
    {
        n = get_int("Positive integer: ");
    }
    while (n < 1);
    int answer = sigma(n);
    printf("%i\n", answer);
}

// Return sum of 1 through m
int sigma(int m)
{
    int sum = 0;
    for (int i = 1; i <= m; i++)
    {
        sum += i;
    }
    return sum;
}

File: 6/src6/weeks/3/sigma0.py
# Sums a range of numbers iteratively


from cs50 import get_int


def main():
    while True:
        n = get_int("Positive integer: ")
        if n > 0:
            break
    answer = sigma(n)
    print(answer)


def sigma(m):
    """Return sum of 1 through m"""
    sum = 0
    for i in range(m + 1):
        sum += i
    return sum


if __name__ == "__main__":
    main()
